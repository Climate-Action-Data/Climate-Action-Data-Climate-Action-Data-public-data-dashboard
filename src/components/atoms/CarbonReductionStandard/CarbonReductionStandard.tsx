import { Flex, Box, Text } from '@chakra-ui/react'
import { ImportantText } from '../ImportantText/ImportantText'
import { Standard } from '@/@types/State'

interface CarbonReductionStandardProps {
  data: Standard[]
}

export const CarbonReductionStandard = (props: CarbonReductionStandardProps): React.JSX.Element => {
  const { data } = props
  const colorChart = [`green.600`, `green.700`, `green.800`]
  return (
    <Flex alignItems={`center`} marginBottom={4}>
      {data.map((standard, idx) => (
        <Box key={`${standard.name}`} textAlign={`center`} marginX={`5px`} flex={1}>
          <ImportantText color={colorChart[idx] ? colorChart[idx] : colorChart[colorChart.length - 1]}>{standard.average}%</ImportantText>
          <Text>{standard.name}</Text>
        </Box>
      ))}
    </Flex>
  )
}
