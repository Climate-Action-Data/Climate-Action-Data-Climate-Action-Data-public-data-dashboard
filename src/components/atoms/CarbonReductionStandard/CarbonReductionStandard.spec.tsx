import { render } from '@testing-library/react'
import { CarbonReductionStandard } from './CarbonReductionStandard'

it(`renders correctly`, () => {
  const { container } = render(
    <CarbonReductionStandard
      data={[
        { name: `VCS`, average: 50 },
        { name: `GCC`, average: 50 },
      ]}
    />,
  )
  expect(container).toMatchSnapshot()
})
