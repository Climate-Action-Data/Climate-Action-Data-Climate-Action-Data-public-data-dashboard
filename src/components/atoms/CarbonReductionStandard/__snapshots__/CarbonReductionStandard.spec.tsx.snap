// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div>
  <div
    class="css-hwbdeb"
  >
    <div
      class="css-whmkqt"
    >
      <span
        class="chakra-text css-of4sfu"
      >
        30
        %
      </span>
      <p
        class="chakra-text css-0"
      >
        VCS
      </p>
    </div>
    <div
      class="css-whmkqt"
    >
      <span
        class="chakra-text css-1tv1yfd"
      >
        20
        %
      </span>
      <p
        class="chakra-text css-0"
      >
        GCC
      </p>
    </div>
    <div
      class="css-whmkqt"
    >
      <span
        class="chakra-text css-o9gfvt"
      >
        10
        %
      </span>
      <p
        class="chakra-text css-0"
      >
        ECO
      </p>
    </div>
  </div>
</div>
`;
